internal/v4/api.go

```diff
@@ -26,23 +26,45 @@ import (
 	"gopkg.in/juju/charmstore.v5-unstable/internal/agent"
 	"gopkg.in/juju/charmstore.v5-unstable/internal/charmstore"
 	"gopkg.in/juju/charmstore.v5-unstable/internal/identity"
+	"gopkg.in/juju/charmstore.v5-unstable/internal/mempool"
 	"gopkg.in/juju/charmstore.v5-unstable/internal/mongodoc"
 	"gopkg.in/juju/charmstore.v5-unstable/internal/router"
 )
 
 var logger = loggo.GetLogger("charmstore.internal.v4")
 
+// reqHandlerPool holds a cache of ReqHandlers to save
+// on allocation time.
```

Comment:
Please let's mention this works because request handlers are put into the pool when closed.
It took me some minutes for me to figure it out.